src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -537,70 +547,136 @@ public static class SystemNamespaceType
                     }
                 }
 
-                return LanguageConstants.String;
+                return new(LanguageConstants.String);
             };
 
-        private static TypeSymbol LoadTextContentTypeBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
+        private static FunctionResult LoadTextContentResultBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
         {
-            if (argumentTypes[0] is not StringLiteralType filePathType)
+            return TryLoadTextContentFromFile(binder, fileResolver, diagnostics,
+                (arguments[0], argumentTypes[0]),
+                arguments.Length > 1 ? (arguments[1], argumentTypes[1]) : null,
+                out var fileContent,
+                LanguageConstants.MaxLiteralCharacterLimit)
+                ? new(new StringLiteralType(fileContent))
+                : new(LanguageConstants.String);
+        }
+
+        private static FunctionResult LoadJsonContentResultBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
+        {
+            string? tokenSelectorPath = null;
+            if (arguments.Length > 1)
             {
-                diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[0]).CompileTimeConstantRequired());
-                return LanguageConstants.String;
+                if (argumentTypes[1] is not StringLiteralType tokenSelectorType)
+                {
+                    diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[1]).CompileTimeConstantRequired());
+                    return new(LanguageConstants.Any);
+                }
+                tokenSelectorPath = tokenSelectorType.RawStringValue;
+            }
+            if (!TryLoadTextContentFromFile(binder, fileResolver, diagnostics,
+                    (arguments[0], argumentTypes[0]),
+                    arguments.Length > 2 ? (arguments[2], argumentTypes[2]) : null,
+                    out var fileContent,
+                    LanguageConstants.MaxJsonFileCharacterLimit))
+            {
+                return new(LanguageConstants.Any);
```

Comment:
Return an empty error type here?

Reply:
TBH I don't get why we return ErrorType since diagnostic is blocking the compilation anyway...

Reply:
See other comment - the error type is relevant outside of just blocking compilation.